examples/czw_butlerEOqueries.ipynb

```diff
@@ -0,0 +1,861 @@
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "butler.getKeys('raw')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Get a list of runs available in this Butler by querying for `raw` data, and asking for the set of `run` values:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "['7086']\n"
+     ]
+    }
+   ],
+   "source": [
+    "runList = butler.queryMetadata('raw', ['run'])\n",
+    "print(runList)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "As expected, this list is a single element long, containing the expected run number.\n",
+    "\n",
+    "From the `visit` key in Butler, we can get a list of \"full\" exposures available in the Butler.  This key is constructed by the Butler during data ingest, and is used to organize data taken at the same time (the time is used to construct the key).  This is also going to return a list, but we will only consider the first element for the time being."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "258333657\n"
+     ]
+    }
+   ],
+   "source": [
+    "dataId = {'run': runList[0]}\n",
+    "visitList = butler.queryMetadata('raw', ['visit'], dataId=dataId)\n",
+    "print(visitList[0])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "From this single visit, determine the single CCD images that comprise that \"full exposure\", using the `detectorName` and `detector` keys.  Get one of those single-device images, and pull the formal device name from the header."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "('S00', 0)\n"
+     ]
+    }
+   ],
+   "source": [
+    "dataId.update({'visit': visitList[0]})\n",
+    "imageList = butler.queryMetadata('raw', ['detectorName', 'detector'], dataId=dataId)\n",
+    "print(imageList[0])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This query returns a DM `Exposure` object, which contains a set of (value, mask, variance) images, a WCS, and associated other metadata.  It can be displayed, but I am skipping this step for now."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "<lsst.afw.image.exposure.exposure.ExposureF object at 0x7f3d3e69bdc0>\n"
+     ]
+    }
+   ],
+   "source": [
+    "dataId.update({'detectorName': imageList[0][0], 'detector': imageList[0][1]})\n",
```

Comment:
My understanding is that conceptually speaking, we should not need and use `detector` as the data ID to retrieve data in Butler. There might be bugs yet to be fixed so it's okay for now here I think. Might worth adding a note.

Reply:
Makes sense.  I've changed the markdown block to note this fact.